In [4]:
# Load the required packages
import numpy as np, scipy as sc, matplotlib as plt, seaborn as sns, pandas as pd, statsmodels as sm, sklearn as sk

# Task 1: Source the Data Set
1. Import the Iris data set from the sklearn.datasets module.
2. Explain, in your own words, what the load_iris() function returns.

In [9]:
# Save the iris dataset in the iris object
iris = sk.datasets.load_iris()

In [20]:
iris.keys()

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

The load_iris() function returns a dictionary-like object which contains multiple obkects, including: the iris dataset and the metadata of the dataset. This keys of this object are 'data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'.

# Task 2: Explore the Data Structure
Print and explain the shape of the data set, the first and last 5 rows of the data, the feature names, and the target classes.

In [21]:
iris.data.shape

(150, 4)

The Iris dataset contains 150 rows (each sample is in a separate row) and 4 collums (each collumn corresponds to sepal length, sepal width, petal length and petal width).

In [24]:
iris.data[0:5,]

array([[5.1, 3.5, 1.4, 0.2],
       [4.9, 3. , 1.4, 0.2],
       [4.7, 3.2, 1.3, 0.2],
       [4.6, 3.1, 1.5, 0.2],
       [5. , 3.6, 1.4, 0.2]])

The dataset above contains info about sepal length, sepal width, petal length and petal width for the first 5 samples in the dataset.

In [32]:
iris.data[-5:len(iris.data),]

array([[6.7, 3. , 5.2, 2.3],
       [6.3, 2.5, 5. , 1.9],
       [6.5, 3. , 5.2, 2. ],
       [6.2, 3.4, 5.4, 2.3],
       [5.9, 3. , 5.1, 1.8]])

The dataset above contains info about sepal length, sepal width, petal length and petal width for the last 5 samples in the dataset.

In [33]:
iris.feature_names

['sepal length (cm)',
 'sepal width (cm)',
 'petal length (cm)',
 'petal width (cm)']

The feature_names object gives the explanation what does each collumn in the dataset represents.

In [35]:
iris.target_names

array(['setosa', 'versicolor', 'virginica'], dtype='<U10')

The target_names shows what sample types are present in the dataset.